src/modules/chat_highlight_blacklist_keywords/index.js

```diff
@@ -144,20 +144,81 @@ class ChatHighlightBlacklistKeywordsModule {
     this.handleHighlightSound = this.handleHighlightSound.bind(this);
   }
 
-  updateDefaultHighlightKeywords() {
-    const value = settings.get(SettingIds.HIGHLIGHT_KEYWORDS);
-    if (Object.keys(value).length > 0) return;
+  validateKeywords() {
+    const highlightKeywordsValue = settings.get(SettingIds.HIGHLIGHT_KEYWORDS);
+    const blacklistKeywordsValue = settings.get(SettingIds.BLACKLIST_KEYWORDS);
+
+    this.updateDefaultHighlightKeywords(highlightKeywords);
+
+    let shouldUpdateHighlightKeywords = false;
+    let shouldUpdateBlacklistKeywords = false;
+
+    const validatedHighlightKeywordsValue = Object.fromEntries(
+      Object.entries(highlightKeywordsValue)
```

Comment:
we probably don't need to complexity of entries here, and your "when to persist" logic can actually be simplified since we know it needs to persist iff legacy types are migrated.


```
for (const keywordsValue of [highlightKeywordsValue, blacklistKeywordsValue]) {
  let updated = false;

  for (const value of Object.values(keywordsValue)) {
    if (![KeywordTypes.EXACT, KeywordTypes.WILDCARD].includes(value.type)) {
      continue;
    }

    if (value.type === KeywordTypes.EXACT) {
      value.keyword = `<${value.keyword}>`;
    }
    value.type = KeywordTypes.MESSAGE;
    updated = true;
  }

  if (updated) {
    settings.set(keywordsValue === highlightKeywordsValue ? SettingIds.HIGHLIGHT_KEYWORDS : SettingIds.BLACKLIST_KEYWORDS, keywordsValue);
  }
}

```